lib/exec.js

```diff
@@ -27,7 +27,11 @@ function exec (cmd, args = [], opts = {}) {
   return new B((resolve, reject) => {
     // spawn the child process with options; we don't currently expose any of
     // the other 'spawn' options through the API
-    let proc = spawn(cmd, args, {cwd: opts.cwd, env: opts.env, shell: opts.shell});
+    let proc = spawn(cmd, args.map(arg => quoteSpawnArgument(arg)), {
```

Comment:
Seems like the helper method ought to be written so the mapping doesn't need to be done outside of it every time it's used.

Reply:
I was thinking about applying the same method to the `cmd` argument

Reply:
Two methods is not the worst thing. Or a method that maps if it's an array, and doesn't if it is not.

Reply:
Implemented the second option